doc/nrf/templates/library_template_README.rst

```diff
@@ -0,0 +1,143 @@
+.. _library_template:
+
+Template: Library
+#################
+
+.. contents::
+   :local:
+   :depth: 2
+
+.. note::
+   Use this template to create pages that describe the libraries included in |NCS|, following these instructions:
+
+   * Use this first section to give a short description (1 or 2 sentences) of the library, indicating what it does and why it should be used.
+
+   * Give the library a concise name that also corresponds to the folder name.
+
+     If the library targets a specific protocol or device, add it in the title before the library name (for example, "NFC:" or "nRF9160:").
+     Do not include the word "library" in the title.
+     Use the provided stock phrases and includes when possible.
+
+   * Sections with ``*`` are optional and can be left out.
+
+     All the other sections are required for all libraries.
+     Do not add new sections or remove a mandatory section without consulting a tech writer.
+     If any new section is needed, try first to fit it as a subsection of one of the sections listed below.
+     When you keep an optional section, remove the asterisk from the title, and format the length of the related RST heading accordingly.
+
+   * Include a tech writer in the PR review.
+
+Overview
+********
+
+.. note::
+   Use this section to give a general overview of the library.
+
+Implementation
+==============
+
+.. note::
+   Use this section to describe the library architecture and implementation.
+
+Supported features
+==================
+
+.. note::
+   Use this section to describe the features supported by the library.
+
+Supported backends*
+===================
+
+.. note::
+   Use this section to describe the backends supported by the library, if applicable.
+   This is optional.
+
+Requirements*
+*************
+
+.. note::
+   Use this section to list the requirements needed to use the library.
+   This is not required if there are no specific requirements to use the library. It is a mandatory section if there are specific requirements that have to be met to use the library.
+
+Library files
```

Comment:
Isn't this already indicated by the source/header files in API documentation?

Reply:
This is supposed to indicate just where to find the files in a more explicit manner. It was one of the original requirements from the JIRA this PR is trying to solve.

Reply:
Really not sure about the added value of this section as the same information is in API documentation as Grzegorz said. I just don't like duplicate information which may be out of sync.
So haven't people been able to find the source file path from the API documentation section or has it not been there in those cases?

Reply:
We got feedback about people not always finding easily the library files. As the PR just got merged, we can continue this discussion in teams.